lib/stats/per-browser.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+const _ = require('lodash');
+
+const Stats = require('./index');
+
+module.exports = class PerBrowserStats extends Stats {
+    constructor(runner) {
+        super(runner);
+
+        this._browsers = {};
+    }
+
+    addRetries(test) {
+        super.addRetries();
+
+        const bro = this._getBrowser(test.browserId);
+
+        bro.retries++;
+    }
+
+    getResult() {
+        const data = super.getResult();
+
+        _.values(this._browsers).forEach((bro) => {
+            bro.total = bro.passed + bro.failed + bro.skipped;
+            delete bro._tests;
+        });
+
+        return _.extend(data, {perBrowser: this._browsers});
+    }
+
+    _addStat(stat, test) {
+        super._addStat(stat, test);
+
+        this._addBrowserStat(stat, test);
+    }
+
+    _addBrowserStat(stat, test) {
+        const bro = this._getBrowser(test.browserId);
+        const key = `${this._buildSuiteKey(test)} ${this._buildStateKey(test)}`;
+        const prevStat = bro._tests[key];
+
+        if (!prevStat) {
```

Comment:
This is almost complete copy-paste from `BaseStats._addStat`.
Not sure if it's a an issue.

Reply:
Don't really like the idea with the third argument.
I suggest we extract a separate method `_buildKey` and another one like `_countStatOnce` (or smth like that) with two arguments that hold `tests`' and `stat` objects.